src/textual/pilot.py

```diff
@@ -104,21 +112,41 @@ async def click(
         else:
             target_widget = screen
 
+        if not target_widget.size.contains(*offset):
+            raise OutOfBounds(
+                f"Target size is {target_widget.size}, click offset is {offset}."
+            )
+
         message_arguments = _get_mouse_message_arguments(
             target_widget, offset, button=1, shift=shift, meta=meta, control=control
         )
+
+        click_offset = Offset(message_arguments["x"], message_arguments["y"])
+        visible_screen_region = screen.region + screen.scroll_offset
+        if not visible_screen_region.contains(*click_offset):
+            raise OutOfBounds(
+                "Target offset is outside of currently-visible"
+                + f"screen region {visible_screen_region}."
```

Comment:
Don't think we need the region repr in there. Can't see it being helpful.

Reply:
Removed.